src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -106,22 +108,37 @@ private ISymbolNode CreateReadyToRunHelperCell(ReadyToRunHelper helperId)
             MethodWithToken method,
             bool isUnboxingStub,
             bool isInstantiatingStub,
+            bool isPrecodeImportRequired,
             SignatureContext signatureContext)
         {
             IMethodNode methodImport;
-            TypeAndMethod key = new TypeAndMethod(method.ConstrainedType, method, isUnboxingStub, isInstantiatingStub);
+            TypeAndMethod key = new TypeAndMethod(method.ConstrainedType, method, isUnboxingStub, isInstantiatingStub, isPrecodeImportRequired);
             if (!_importMethods.TryGetValue(key, out methodImport))
             {
                 if (CompilationModuleGroup.ContainsMethodBody(method.Method, false))
```

Comment:
Hmm, can you please explain to me why we don't need similar logic for methods outside of the current module?